### PR TITLE
Eip1559 state tests

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

**Review by marcindsobczak — APPROVED** [review]

File: src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs [hunk 1]
```diff
@@ -103,7 +104,13 @@ protected EthereumTestResult RunTest(GeneralStateTest test, ITxTracer txTracer)
             header.Hash = header.CalculateHash();
 
             Stopwatch stopwatch = Stopwatch.StartNew();
-            transactionProcessor.Execute(test.Transaction, header, txTracer);
+            var txValidator = new TxValidator((MainnetSpecProvider.Instance.ChainId));
+            var spec = specProvider.GetSpec(test.CurrentNumber);
+            if (test.Transaction.ChainId == null)
+                test.Transaction.ChainId = MainnetSpecProvider.Instance.ChainId;
+            bool isValid = txValidator.IsWellFormed(test.Transaction, spec);
+            if (isValid)
```
Comment (tkstanczak) [inline src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:112]: I think you need to throw on invalid? or not?

**Review by tkstanczak — APPROVED** [review]

File: src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:112]: I think not, Ethereum tests are passing now, this was actually added by @MarekM25

Comment (MarekM25) [reply src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:112]: Another solution is adding TxValidator logic in TransactionProcessor. What do you think @tkstanczak @LukaszRozmej ?

Comment (tkstanczak) [reply src/Nethermind/Ethereum.Test.Base/GeneralTestBase.cs:112]: ignore

**Review by tkstanczak — APPROVED** [review]

**Pull request merged**